hr_planning/models/resource_calendar.py

```diff
@@ -12,27 +12,28 @@
 from odoo import api, models
 
 
-class resource_calendar(models.Model):
+class ResourceCalendar(models.Model):
     _inherit = 'resource.calendar'
 
     @api.model
     def create(self, vals):
-        res = super(resource_calendar, self).create(vals)
+        res = super(ResourceCalendar, self).create(vals)
 
         if ('attendance_ids' in vals):
             res._generate()
         return res
 
     @api.multi
     def write(self, vals):
-        res = super(resource_calendar, self).write(vals)
+        res = super(ResourceCalendar, self).write(vals)
 
         if ('attendance_ids' in vals):
             self._generate()
         return res
 
-    @api.one
+    @api.multi
     def _generate(self):
+        self.ensure_one()
```

Comment:
You should better add a loop through the records in self. I see at line 31 that the method could be called on several records.